src/core/ddsc/tests/psmx.c

```diff
@@ -98,6 +99,67 @@ static dds_entity_t create_participant (dds_domainid_t int_dom)
   return pp;
 }
 
+static void config_psmx_free_content(struct ddsi_config_psmx* cfg)
+{
+  assert(cfg);
+  ddsrt_free(cfg->name);
+  ddsrt_free(cfg->library);
+  ddsrt_free(cfg->config);
+}
+
+static void config_psmx_elm_free(struct ddsi_config_psmx_listelem* elm)
+{
+  assert(elm);
+  config_psmx_free_content(&elm->cfg);
+  ddsrt_free(elm);
+}
+
+static bool domain_pin(dds_entity_t domain, dds_domain** dom_out)
+{
+  dds_entity* x = NULL;
+  dds_return_t rc = dds_entity_pin(domain, &x);
+  if ( rc == DDS_RETCODE_OK && dds_entity_kind(x) == DDS_KIND_DOMAIN ) {
+    *dom_out = (dds_domain*)x;
+    return true;
+  }
+  return false;
```

Comment:
This should unpin the entity if `rc == DDS_RETCODE_OK`

Reply:
The unpin happens in `domain_unpin()`.

Reply:
But `domain_unpin` is not called in case this function returns `false` because the entity kind is incorrect.